lib/quicly.c

```diff
@@ -1947,19 +2020,18 @@ int retire_acks(quicly_conn_t *conn, size_t count, size_t epoch)
     do {
         if ((pn = ack->packet_number) == UINT64_MAX)
             break;
-        do {
-            int skip = 0;
-            if (epoch != SIZE_MAX) {
-                if (ack_is_handshake_flow(ack, epoch)) {
-                    skip = 0;
-                } else if (epoch == 1 &&
-                           (ack->acked == on_ack_stream && !stream_is_handshake_flow(ack->data.stream.stream_id, 3))) {
-                    /* NACK all application data upon 0-RTT denial */
-                    skip = 0;
-                } else {
-                    skip = 1;
-                }
+        int skip = 0;
+        if (epoch != SIZE_MAX) {
```

Comment:
I don't think I follow this block below. What does this have to do with congestion control?

Reply:
We are (still) using a unified inflight packet map (i.e. `quicly_acks_t`).

This means that when certain epoch is being closed, we need to remove references to the packets sent in that epoch from the map. The other point is that when 0-RTT is rejected we want to pretend as if we received NACK for all data we sent using 0-RTT.

This function does that, when `epoch` is used (i.e. set to something other than `SIZE_MAX`).

Reply:
I see SIZE_MAX used all around this project (and picotls) as a "None" value.
Why don't you just define a "NONE" constant for that? It would make these expressions easier to understand.